framework/src/utils/MooseCoordTransform.C

```diff
@@ -399,43 +409,49 @@ MooseCoordTransform::operator()(const Point & point) const
 }
 
 Point
-MooseCoordTransform::mapBack(const Point & point) const
+MultiCoordTransform::mapBack(const Point & point) const
 {
   Point ret(point);
 
   // inverse translate
   ret -= _translation;
 
   // inverse rotate and then inverse scale
-  if (_rs_inverse)
-    ret = (*_rs_inverse) * ret;
+  if (_single_app_transform._rs_inverse)
+    ret = (*_single_app_transform._rs_inverse) * ret;
+
+  if (_skip_coordinate_collapsing)
+    return ret;
 
   // Finally, coordinate system conversions
-  if ((_coord_type == Moose::COORD_XYZ && (_destination_coord_type == Moose::COORD_RZ ||
-                                           _destination_coord_type == Moose::COORD_RSPHERICAL)) ||
-      (_coord_type == Moose::COORD_RZ && _destination_coord_type == Moose::COORD_RSPHERICAL))
+  if ((_single_app_transform._coord_type == Moose::COORD_XYZ &&
+       (_destination_coord_type == Moose::COORD_RZ ||
+        _destination_coord_type == Moose::COORD_RSPHERICAL)) ||
+      (_single_app_transform._coord_type == Moose::COORD_RZ &&
+       _destination_coord_type == Moose::COORD_RSPHERICAL))
     mooseError("Coordinate collapsing occurred in going to the reference space. There is no unique "
                "return mapping");
 
   return ret;
 }
 
 void
-MooseCoordTransform::setTranslationVector(const Point & translation)
+MultiCoordTransform::setTranslationVector(const Point & translation)
 {
   _translation = translation;
 }
 
 void
-MooseCoordTransform::setDestinationCoordinateSystem(
-    const MooseCoordTransform & destination_coord_transform)
+MultiCoordTransform::setDestinationCoordinateSystem(
+    const MultiCoordTransform & destination_coord_transform)
 {
-  _destination_coord_type = destination_coord_transform._coord_type;
-  _destination_r_axis = destination_coord_transform._r_axis;
-  _destination_z_axis = destination_coord_transform._z_axis;
+  _destination_coord_type = destination_coord_transform._single_app_transform._coord_type;
```

Comment:
this makes me feel that a Multi(App)CoordTransform should have two attributes:
- origin single-app-transform
- desitrination single-app-transform
and work with those

Reply:
:+1: